azure-iot-device/azure/iot/device/common/transport/mqtt/mqtt_transport.py

```diff
@@ -334,17 +345,24 @@ def _on_provider_message_received_callback(self, topic, payload):
         :param bytes payload: Payload of the message as a bytestring
         """
         logger.info("Message received on topic {}".format(topic))
-        message_received = Message(payload)
         topic_parts = topic.split("/")
 
         if _is_input_topic(topic):
+            message_received = Message(payload)
             input_name = topic_parts[TOPIC_POS_INPUT_NAME]
             message_received.input_name = input_name
-            _extract_properties(topic_parts[TOPIC_POS_MODULE], message_received)
+            _extract_message_properties(topic_parts[TOPIC_POS_MODULE], message_received)
             self.on_transport_input_message_received(input_name, message_received)
         elif _is_c2d_topic(topic):
-            _extract_properties(topic_parts[TOPIC_POS_DEVICE], message_received)
+            message_received = Message(payload)
+            _extract_message_properties(topic_parts[TOPIC_POS_DEVICE], message_received)
             self.on_transport_c2d_message_received(message_received)
+        elif _is_method_topic(topic):
+            rid = topic_parts[TOPIC_POS_REQUEST_ID].split("=")[1]
```

Comment:
>TOPIC_POS_REQUEST_ID [](start = 30, length = 20)

it's dangerous to function on position for properties going after the '?' character. in fact we just had a bug about that in the node DPS sdk.
what needs to happen is everything after the '?' must be treated as key/value pairs that could appear in any order.
I would suggest adding a fix for this before merging, or fix it immediately after.